api/src/data/models/data-ingestion-mapping-model.ts

```diff
@@ -0,0 +1,8 @@
+export interface DataIngestionMapping {
+  id?: number;
```

Comment:
Consider using Sequelize, rather than having an "optional" ID column.
Or investigate having:
1. type for attributes required for creation
2. type for model that has been created and loaded from the database.